Source/WebCore/platform/graphics/nicosia/NicosiaImageBufferPipe.cpp

```diff
@@ -62,60 +69,85 @@ NicosiaImageBufferPipeSource::~NicosiaImageBufferPipeSource()
-                {
-                    Locker locker { m_imageBufferLock };
-
-                    if (!m_imageBuffer)
-                        return;
+                auto nativeImage = WTFMove(m_image);
 
-                    auto nativeImage = ImageBuffer::sinkIntoNativeImage(WTFMove(m_imageBuffer));
-                    if (!nativeImage)
-                        return;
+                auto size = nativeImage->size();
+                OptionSet<BitmapTexture::Flags> flags;
+                if (nativeImage->hasAlpha())
+                    flags.add(BitmapTexture::Flags::SupportsAlpha);
+                texture = BitmapTexture::create(size, flags);
 
-                    auto size = nativeImage->size();
-                    OptionSet<BitmapTexture::Flags> flags;
-                    if (nativeImage->hasAlpha())
-                        flags.add(BitmapTexture::Flags::SupportsAlpha);
-                    texture = BitmapTexture::create(size, flags);
 #if USE(CAIRO)
-                    auto* surface = nativeImage->platformImage().get();
-                    auto* imageData = cairo_image_surface_get_data(surface);
-                    texture->updateContents(imageData, IntRect(IntPoint(), size), IntPoint(), cairo_image_surface_get_stride(surface));
+                auto* surface = nativeImage->platformImage().get();
+                auto* imageData = cairo_image_surface_get_data(surface);
+                texture->updateContents(imageData, IntRect(IntPoint(), size), IntPoint(), cairo_image_surface_get_stride(surface));
 #elif USE(SKIA)
-                    auto* image = nativeImage->platformImage().get();
-                    // FIXME: support accelerated offscreen canvas.
-                    RELEASE_ASSERT(!image->isTextureBacked());
+                auto image = nativeImage->platformImage();
+                if (image->isTextureBacked()) {
+                    fence->wait(WebCore::GLFence::FlushCommands::No);
+                    texture->copyFromExternalTexture(textureID);
+                    fence = GLFence::create();
+                } else {
                     SkPixmap pixmap;
                     if (image->peekPixels(&pixmap))
                         texture->updateContents(pixmap.addr(), IntRect(IntPoint(), size), IntPoint(), image->imageInfo().minRowBytes());
-#endif
                 }
+#endif
+            }
 
-                auto layerBuffer = makeUnique<TextureMapperPlatformLayerBuffer>(WTFMove(texture));
-                layerBuffer->setExtraFlags(TextureMapperFlags::ShouldBlend);
-                downcast<TextureMapperPlatformLayerProxyGL>(proxy).pushNextBuffer(WTFMove(layerBuffer));
-            });
-        };
-        proxyOperation(m_nicosiaLayer->proxy());
-    }
+            auto layerBuffer = makeUnique<TextureMapperPlatformLayerBuffer>(WTFMove(texture));
+            layerBuffer->setExtraFlags(TextureMapperFlags::ShouldBlend);
+            layerBuffer->setFence(WTFMove(fence));
```

Comment:
This method is guarded with `#if PLATFORM(GTK) || PLATFORM(WPE)` so this patch broke PlayStation.